news-bundle/src/Controller/NewsFeedController.php

```diff
@@ -0,0 +1,281 @@
+        $contentModel = $this->framework->getAdapter(ContentModel::class);
+
+        $description = $model->teaser ?? '';
+
+        // Prepare the description
+        if ('source_text' === $feedSource) {
+            $description = '';
+            $elements = $contentModel->findPublishedByPidAndTable($model->id, 'tl_news');
+
+            if (null !== $elements) {
+                // TODO: Is this still necessary?
+                // Overwrite the request (see #7756)
+                $requestUri = $environment->get('request');
+                $environment->set('request', $item->getLink());
+
+                while ($elements->next()) {
+                    $description .= $controller->getContentElement($elements->current());
+
+                    if ($this->responseTagger) {
+                        $this->responseTagger->addTags(['contao.db.tl_content.'.$elements->id]);
+                    }
+                }
+
+                $environment->set('request', $requestUri);
+            }
+        }
+
+        $description = $this->insertTags->replaceInline($description);
+
+        return $controller->convertRelativeUrls($description, $item->getLink());
+    }
+
+    private function addEnclosures(Item $item, array|string $enclosures, string $imageSize = null): void
+    {
+        if (\is_string($enclosures)) {
+            $enclosures = [$enclosures];
+        }
+
+        $size = StringUtil::deserialize($imageSize, true);
+
+        $filesAdapter = $this->framework->getAdapter(FilesModel::class);
+        $files = $filesAdapter->findMultipleByUuids($enclosures);
+
+        if (null === $files) {
+            return;
+        }
+
+        $staticUrl = $this->contaoContext->getStaticUrl();
+        $baseUrl = $staticUrl ?: $this->request->getSchemeAndHttpHost();
+
+        while ($files->next()) {
+            $file = new File($files->path);
+
+            $fileUrl = $baseUrl.'/'.$file->path;
+            $fileSize = $file->filesize;
+
+            if ($size && $file->isImage) {
+                $image = $this->imageFactory->create(Path::join($this->projectDir, $file->path), $size);
+                $fileUrl = $baseUrl.'/'.$image->getUrl($this->projectDir);
+                $file = new File(Path::makeRelative($image->getPath(), $this->projectDir));
+                $fileSize = $file->exists() ? $file->filesize : null;
+            }
+
+            $media = new Media();
+            $media->setUrl($fileUrl)
+                ->setType($file->mime)
+            ;
+
+            if ($fileSize) {
+                $media->setLength($fileSize);
+            }
+
+            $item->addMedia($media);
+        }
+    }
+
+    /**
+     * Set the cache headers according to the page settings.
+     *
+     * @param Response $response The response object
+     *
+     * @return Response The response object
+     */
+    private function setCacheHeaders(Response $response, PageModel $pageModel)
```

Comment:
Wooooh, this looks very much like something we could use in other Contao controllers too! What do you think about moving that to `AbstractController`? :)

Reply:
Or even the `EntityCacheTags` helper?